libraries/SuplaDevice/examples/HC_SR04_Distance_sensor_extended/HC_SR04_Distance_sensor_extended.ino

```diff
@@ -0,0 +1,87 @@
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+#include <SuplaDevice.h>
+
+// Add include to HC_SR04 sensor
+#include <supla/sensor/HC_SR04.h>
+
+// Choose proper network interface for your card:
+// Arduino Mega with EthernetShield W5100:
+#include <supla/network/ethernet_shield.h>
+// Ethernet MAC address
+uint8_t mac[6] = {0x00, 0x01, 0x02, 0x03, 0x04, 0x05};
+Supla::EthernetShield ethernet(mac);
+//
+// Arduino Mega with ENC28J60:
+// #include <supla/network/ENC28J60.h>
+// Supla::ENC28J60 ethernet(mac);
+//
+// ESP8266 based board:
+// #include <supla/network/esp_wifi.h>
+// Supla::ESPWifi wifi("your_wifi_ssid", "your_wifi_password");
+//
+// ESP32 based board:
+// #include <supla/network/esp32_wifi.h>
+// Supla::ESP32Wifi wifi("your_wifi_ssid", "your_wifi_password");
+
+int8_t trigPin = 13;
+int8_t echoPin = 12;
+int16_t minSensorRead =  5; //minimum sensor reading distance in centimeters
+int16_t maxSensorRead = 55; //maximum sensor reading distance in centimeters
+int16_t minAppReading = 50; //minimum distance shown by the App in centimeters when sensor read <= min_sensor_read
+int16_t maxAppReading =  0; //maximum distance shown by the App in centimeters when sensor read >= max_sensor_read
+
+void setup() {
+
+  Serial.begin(115200);
+
+  // Replace the falowing GUID with value that you can retrieve from https://www.supla.org/arduino/get-guid
+  char GUID[SUPLA_GUID_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
+
+  // Replace the following AUTHKEY with value that you can retrieve from: https://www.supla.org/arduino/get-authkey
+  char AUTHKEY[SUPLA_AUTHKEY_SIZE] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};
+
+  /*
+   * Having your device already registered at cloud.supla.org,
+   * you want to change CHANNEL sequence or remove any of them,
+   * then you must also remove the device itself from cloud.supla.org.
+   * Otherwise you will get "Channel conflict!" error.
+   */
+
+  new Supla::Sensor::HC_SR04(trigPin,echoPin, minSensorRead, maxSensorRead, minAppReading, maxAppReading); 
+ 
+  // Supla::Sensor::HC_SR04(trigPin, echoPin) // sends sensor reading to cloud unchanged
+
+
+  /*
+   * SuplaDevice Initialization.
+   * Server address, LocationID and LocationPassword are available at https://cloud.supla.org 

```

Comment:
this comment is not valid anymore. You can copy from other examples current version